lib/proxyquire.js

```diff
@@ -129,17 +130,47 @@ Proxyquire.prototype.load = function (request, stubs) {
   return this._withoutCache(this._parent, stubs, request, this._parent.require.bind(this._parent, request))
 }
 
+// Resolves a stub relative to a module.  Note that this only resolves
+// "relative" paths (e.g. "./foo") but not non-relative modules (e.g. "react").
+// Also, if noPreserveCache(), then we won't throw if the module can't
+// be resolved.
+Proxyquire.prototype._resolveRelativePath = function (path, moduleFilename) {
+  try {
+    return resolve.sync(path, {
+      basedir: dirname(moduleFilename),
+      extensions: Object.keys(require.extensions),
+      paths: Module.globalPaths
+    })
+  } catch (err) {
+    // If this is not a relative path (e.g. "foo" as opposed to "./foo"), and
+    // we couldn't resolve it, then we just let the path through unchanged.
+    if (path[0] !== '.') {
```

Comment:
Thoughts on using https://github.com/mattdesl/relative-require-regex here?

IMO an absolute path that couldn't be resolved should also pass through. External module also has the benefits of separate tests and docs.

Reply:
Could do - it's not a very exciting module: https://github.com/mattdesl/relative-require-regex/blob/master/index.js  :P  I tend to lean towards not using dependencies if I can avoid them.

The only thing different between their version and mine is, they would count something that starts with a "/" as a relative path.  We're only going to hit this case when the path doesn't exist, so it's already a super rare case.  Really this is here to keep behavior consistent with what used to happen (and to make the unit tests pass, since they bring in global modules that don't exist).

If you have strong feelings about it one way or the other, though, I'm happy to go either way.

Reply:
We aren't really trying to reproduce node behavior here, though.  Any path that starts with a "." or ".." is unsafe for us to let through, because it wasn't resolved, but it could resolve relative to some other source file (that's exactly the issue I raised in the first place).  A file that starts with a "/", on the other hand, can only match one file on the disk, so it's safe to let through even if the file doesn't exist.  We just want to make sure that anything we return from this function is a "unique" module name.

Reply:
Oh...  And if I change this to use relative-require-regex, two existing tests fail:

<img width="796" alt="screen shot 2018-03-01 at 9 39 54 pm" src="https://user-images.githubusercontent.com/1771003/36880894-297704f6-1d99-11e8-9d2b-66112d618bd8.png">

So these tests would need some rework...  We could add the `noPreserveCache()` setting to these tests, so it falls into the second case, I suppose.

Reply:
Cool. Keep it is as for now. I'll look to remove that path eventually.